src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -338,7 +344,7 @@ protected static bool IdentifiersMatch(ISyntaxFactsService syntaxFacts, string n
                 }
 
                 // 3) If the symbols ARE equivalent ignoring assemblies, they may or may not be equivalent if containing assemblies are NOT ignored.
-                if (equivalentTypesWithDifferingAssemblies.Object.Any())
```

Comment:
We really should add an analyzer for this.  We keep finding latent instances where we're throwing away perf.